src/impl/epoll/external_commit_helper.hpp

```diff
@@ -53,22 +57,38 @@ class ExternalCommitHelper {
         FdHolder(FdHolder const&) = delete;
     };
 
-    void listen();
+    class DaemonThread {
```

Comment:
After tired again i remembered why i didn't do it at the first place. `DaemonThread` also needs `FdHolder` which is declared as a private inner class -- I failed to get it compiled :( I could move the `FdHolder` outside and `DaemonThread` to the cpp file, but that also exposes `FdHolder`. What do you prefer? @bdash